contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,906 @@
+    function withdrawProtocolFee(uint256 _amount) external override {
+        UtilLib.onlyManagerRole(msg.sender, staderConfig);
+        accrueFee();
+        if (_amount > accumulatedProtocolFee) {
+            revert InvalidWithdrawAmount();
+        }
+        accumulatedProtocolFee -= _amount;
+        if (!IERC20(staderConfig.getStaderToken()).transfer(staderConfig.getStaderTreasury(), _amount)) {
+            revert SDTransferFailed();
+        }
+        emit WithdrawnProtocolFee(_amount);
+    }
+
+    /// @notice for max approval to SD collateral contract for spending SD tokens
+    function maxApproveSD() external override whenNotPaused {
+        UtilLib.onlyManagerRole(msg.sender, staderConfig);
+        address sdCollateral = staderConfig.getSDCollateral();
+        UtilLib.checkNonZeroAddress(sdCollateral);
+        IERC20(staderConfig.getStaderToken()).approve(sdCollateral, type(uint256).max);
+    }
+
+    /**
+     * @notice Applies accrued fee to total utilized and protocolFee
+     * @dev This calculates fee accrued from the last check pointed block
+     *   up to the current block and writes new checkpoint to storage.
+     */
+    function accrueFee() public override whenNotPaused {
+        /* Remember the initial block number */
+        uint256 currentBlockNumber = block.number;
+
+        /* Short-circuit accumulating 0 fee */
+        if (accrualBlockNumber == currentBlockNumber) {
+            return;
+        }
+
+        /* Calculate the number of blocks elapsed since the last accrual */
+        uint256 blockDelta = currentBlockNumber - accrualBlockNumber;
+
+        /*
+         * Calculate the fee accumulated into utilized and totalProtocolFee and the new index:
+         *  simpleFeeFactor = utilizationRate * blockDelta
+         *  feeAccumulated = simpleFeeFactor * totalUtilizedSD
+         *  totalUtilizedSDNew = feeAccumulated + totalUtilizedSD
+         *  totalProtocolFeeNew = feeAccumulated * protocolFeeFactor + totalProtocolFee
+         *  utilizeIndexNew = simpleFeeFactor * utilizeIndex + utilizeIndex
+         */
+
+        uint256 simpleFeeFactor = utilizationRatePerBlock * blockDelta;
+        uint256 feeAccumulated = (simpleFeeFactor * totalUtilizedSD) / DECIMAL;
+        totalUtilizedSD += feeAccumulated;
+        accumulatedProtocolFee += (protocolFeeFactor * feeAccumulated) / DECIMAL;
+        utilizeIndex += (simpleFeeFactor * utilizeIndex) / DECIMAL;
+
+        accrualBlockNumber = currentBlockNumber;
+
+        emit AccruedFees(feeAccumulated, accumulatedProtocolFee, totalUtilizedSD);
+    }
+
+    /**
+     * @notice Initiates the liquidation process for an account if its health factor is below the required threshold.
+     * @dev The function checks the health factor, accrues fees, updates utilized indices, and calculates liquidation amounts.
+     * @param account The address of the account to be liquidated
+     */
+    function liquidationCall(address account) external override {
```

Comment:
Can we comment that this flow does not touch operator's self-bonded SD even if that could be used to pay back?